src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -647,18 +673,51 @@ object LazyList extends LazyListFactory[LazyList] {
       this
     }
 
-    protected def tailDefined: Boolean = tlEvaluated
-    protected def headDefined: Boolean = hdEvaluated
+    protected[immutable] def tailDefined: Boolean = tlEvaluated
+    protected[immutable] def headDefined: Boolean = hdEvaluated
   }
 
+  @SerialVersionUID(3L)
+  final class Suspended[A](var eval: () => LazyList[A]) extends LazyList[A] {
+    private[this] var evaluated: Boolean = false
```

Comment:
I have no idea. I _think_ maybe, although since `head`'s protected by its own volatile bitmap field, the danger is that one thread will call `head`, and the other thread will get `false` from `headDefined`, which is unfortunate but probably not harmful. 

I think the `evaluated = true` should go after the call to `eval`, though, now that I think on it.

Reply:
I've been thinking about this a bunch, and I *think* that, because it's not volatile, the JMM allows another thread to read a value of `true` from `evaluated`, as long as it gets set to `true` at some point (possibly only if set by another thread?). Basically, without `@volatile`, there is no happens-before relationship between the write of `true` and the read of `true` if they (might?) happen on different threads, so it might not happen before.